### PR TITLE
Add aten::select operator and layer tests

### PR DESCRIPTION
**Pull request by sikorsl1**
Details:

- `aten::select` operator has been added
- layer test for this operator has been added

Ticket:

- 94267

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_select.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@pytest.mark.parametrize('input_dim', list(range(4)))
+@pytest.mark.parametrize('input_index', list(range(4)))
+class TestSelect(PytorchLayerTest):
+
+    def _prepare_input(self):
+        return (np.random.randn(4, 4, 5, 5).astype(np.float32),)
+
+    def create_model(self):
+
+        input_dim = self.input_dim
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_select.py]: Please pass `input_dim` and `input_index` as parameters to `create_model` and `aten_select` init.

File: tests/layer_tests/pytorch_tests/test_select.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from pytorch_layer_test_class import PytorchLayerTest
+import numpy as np
+import torch
+
+
+@pytest.mark.parametrize('input_dim', list(range(4)))
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_select.py]: Can dim and index be negative? If yes, please add tests

Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_select.py]: Right, it can be a negative number! Good catch, thanks for this comment :) I added proper test data to the test.

File: tests/layer_tests/pytorch_tests/test_select.py (hunk 1 above)
Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_select.py]: I added `input_dim` and `input_index` arguments to the `create_model` method. Since I can use them as `nonlocal` variables in `aten_select` I don't think that it is required to put them explicite in the `__init__` method.

Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_select.py]: I don't like `nonlocal` variables, if possible I'd rather avoid that.

Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_select.py]: No problem, I made it as you suggested.

File: src/frontends/pytorch/src/op/select.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    auto input_tensor = context.get_input(0);
+    auto dim = std::make_shared<opset8::Reshape>(context.get_input(1), const_1, false);
+    auto start = std::make_shared<opset8::Reshape>(context.get_input(2), const_1, false);
+
+    auto less = std::make_shared<opset8::Less>(start, const_0);
+    auto const_1_signed = std::make_shared<opset8::Select>(less, const_minus_1, const_1);
+    auto stop = std::make_shared<opset8::Add>(start, const_1_signed);
+
+    auto slice_node = std::make_shared<opset8::Slice>(input_tensor, start, stop, const_1_signed, dim);
+    context.mark_node(slice_node);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/select.cpp]: Please call `mark_node` or `mark_nodes` for all other nodes you create here including consts

Comment (sikorsl1) [reply src/frontends/pytorch/src/op/select.cpp]: done

File: tests/layer_tests/pytorch_tests/test_select.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    def _prepare_input(self):
+        return (np.random.randn(4, 4, 5, 5).astype(np.float32),)
+
+    def create_model(self, input_dim, input_index):
+
+        class aten_select(torch.nn.Module):
+
+            def __init__(self, input_dim, input_index) -> None:
+                super().__init__()
+                self.dim = torch.from_numpy(
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_select.py]: You do not need to create torch tensor here, because in `forward` you use it just as int. Please remove that from `dim` and `index`. Other then that everything is good.

Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_select.py]: Sure, good idea :) I have corrected it.

**Review by mvafin — APPROVED** [review]
